presto-main/src/main/java/com/facebook/presto/metadata/PolymorphicScalarFunction.java

```diff
@@ -173,4 +177,47 @@ private MethodHandle applyExtraParameters(Method matchingMethod, List<Object> ex
         }
         return clazz;
     }
+
+    public static final class PolymorphicScalarFunctionChoice
+    {
+        private final boolean nullableResult;
+        private final List<ArgumentProperty> argumentProperties;
+        private final List<MethodsGroup> methodsGroups;
+
+        public PolymorphicScalarFunctionChoice(
+                boolean nullableResult,
+                List<ArgumentProperty> argumentProperties,
+                List<MethodsGroup> methodsGroups)
+        {
+            this.nullableResult = nullableResult;
+            this.argumentProperties = ImmutableList.copyOf(requireNonNull(argumentProperties, "argumentProperties is null"));
+            this.methodsGroups = ImmutableList.copyOf(requireNonNull(methodsGroups, "methodsWithExtraParametersFunctions is null"));
+        }
+
+        private int getNumberOfBlockPositionArguments(List<ArgumentProperty> argumentProperties)
```

Comment:
`getNumberOfBlockPositionArguments` is unused (getters are also unused currently, but once you address my comments above they won't be).